llms/googleai/internal/palmclient/palmclient.go

```diff
@@ -48,19 +42,20 @@ type PaLMClient struct {
 }
 
 // New returns a new Vertex AI based PaLM API client.
-func New(projectID string, opts ...option.ClientOption) (*PaLMClient, error) {
+func New(ctx context.Context, projectID, location string, opts ...option.ClientOption) (*PaLMClient, error) {
 	numConns := runtime.GOMAXPROCS(0)
 	if numConns > defaultMaxConns {
 		numConns = defaultMaxConns
 	}
 	o := []option.ClientOption{
 		option.WithGRPCConnectionPool(numConns),
-		option.WithEndpoint(defaultAPIEndpoint),
+		option.WithEndpoint(fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)),
```

Comment:
@eliben I felt the previous endpoint is hard code and couldn't be set, possibly due to a bug, so I modified it.

Reply:
I suggest making fewer changes at once, to make the PR easier to review?
Can you please adjust the PR to just add the new option? You can sync with head - it should have the latest go.mod/go.sum changes, so please don't include those in the PR either

Reply:
Understood. I have already been using this branch in the production environment. I will make the necessary adjustments when I find some time.